runtime/Go/antlr/atn_config_set.go

```diff
@@ -155,14 +158,17 @@ func (b *BaseATNConfigSet) GetStates() *Set {
 	return states
 }
 
+// HasSemanticContext returns true if this config has a semantic context.
```

Comment:
I like your "whether" wording in SetHasSemanticContext below, it explains both Boolean cases. Here, the doc only explains what happens if it's true.